### DISCUSSION TITLE
Select folders to scan

### DISCUSSION THREAD

**Discussion by Shamaoke** [deckerst/aves#134] [Q&A] [answered]
Hi.

When Aves performs a scan for content, absolutely all image files found in the user's folder are added to the collection. This includes photos of documents, cd-covers, UI elements of some applications and other similar staff. In my case the quantity of such irrelevant images exceeds the quantity of ordinary photos and videos which I'd like Aves to manage.

I'd like to be able to control in which exact folders the application will search for content and to not allow it from searching in other ones.

Is it feasible to achieve?

Thanks.

**Comment by deckerst** [comment, marked as answer]
It's possible to exclude folders and paths, but not to only include some.

In your case, maybe a couple of excluded paths would be enough. For example, to exclude all the CD covers spread across your music collection, you could just exclude the root of it (something like `/storage-volume/Music/`).

You can check out the `Settings` > `Privacy` > `Hidden paths` to select paths to exclude.

**Comment by Shamaoke** [comment]
1. In `Settings → Privacy → Hidden paths` the information section states that `Photos and videos in these folders, or any of their subfolders, will _not appear_ in your collection.`, however, when tapping the plus button, selecting a folder and tapping the “Use this folder” button, the information popup asks `Allow Aves to _access_ files in Folder?` and then `This will let Aves _access_ current and future content stored in Folder`. This controversion is totally misleading.

2. If you have several hundred folders in the user's home folder, it's extremely tedious to select them all.

3. When new folders are created in the user's home folder, you will constantly have to remember to include these folders in the filter.

4. For some reason the `/storage/emulated/0/Documents` folder can't be added to the filter.

I have a couple more questions.

Will the quantity and quality of images affect the performance of the application and memory consumption? E. g. if I have 10 or 10 000 images, what impact on performance and memory consumption will it have?

What the program actually do when I include folders in the Hidden paths filter? Does it just cosmetically hide them but still continues to work as if them were not included in the filter?

Thanks.

_Android 10_; _Aves 1.5.6 (app-izzy-arm64-v8a-release.apk)_

**Reply by deckerst** [reply]
1. Correct. What I should actually do is provide a way for the user to just browse and select a directory. I didn't feel like reimplementing/researching this at the time, so I simply reused the OS directory picker but, as you noticed, this is not the proper way.
2. Sure, but do you have hundred folders in your home? Android is pushing for a structure where documents are in Documents, music in Music, etc. so that people don't end up with a hundred folders in their home.
3. If new folders are created down within hidden subtrees, these new folders are already taken care of (like my earlier example of CD covers in a music collection and hiding the root `[volume]/Music/`).
4. Indeed, this is because of 1. I should provide a different way to pick directories.

> Will the quantity and quality of images affect the performance of the application and memory consumption? E. g. if I have 10 or 10 000 images, what impact on performance and memory consumption will it have?

Quantity will impact loading time and memory. I use the app on my own device with about 13000 items. With that collection, the loading time and memory are acceptable, but I'm interested in feedback from people with 50000+ items.

Quality shouldn't impact loading time or browsing the collection. I tested the viewer with images over 2GB+ and it's fine (thanks to tiling/resampling). The viewer struggles a bit with heavy multi-pages HEIF images (like the 100MB+ macOS wallpapers with 16 day/night variants). Thumbnails of heavier images will also take a longer time to load (if they're not already cached).

> What the program actually do when I include folders in the Hidden paths filter? Does it just cosmetically hide them but still continues to work as if them were not included in the filter?

Hiding, whether by path of other filter, only impacts what is displayed within the app. It does not hide them from the Android Media Store (via `.nomedia` and such), nor does it move the files or edit them (adding tags or other metadata). This is purely internal to the app. Files that match a hidden filter are only processed as little as necessary. For example:
- if a file is hidden by path, the file is not processed at all because the file path is already known,
- if a file is hidden by country, the file has to be first catalogued and reverse geocoded before Aves knows whether the country for that file matches the filter.

**Reply by deckerst** [reply]
You can subscribe to #136 for progress on 1. and 4.

**Reply by Shamaoke** [reply]
I do. Thanks.

**Reply by deckerst** [reply]
In Aves v1.5.7 you can select hidden paths with a different dialog.

I'll mark this thread as answered, but feel free to ask if you more questions.
